tests/test_unauthenticated_reddit.py

```diff
@@ -103,6 +106,48 @@ def test_get_front_page(self):
         num = 50
         self.assertEqual(num, len(list(self.r.get_front_page(limit=num))))
 
+    def test_load_config(self):
+        from praw import settings
+        environkeys = ('APPDATA', 'XDG_CONFIG_HOME', 'HOME')
+        environkeys = {key: os.getenv(key) for key in environkeys}
+        knownkey = list(filter(None, environkeys.values()))
+        knownkey = knownkey[0] if knownkey else None
+        for environkey, oldkey in environkeys.items():
+            os.environ[environkey] = knownkey
+            try:
+                reload_module(settings)
+            except Exception:
+                raise AssertionError("Could not load config "
+                                     "for key {}".format(environkey))
+            finally:
+                del os.environ[environkey]
+                if oldkey is not None:
+                    os.environ[environkey] = oldkey
+        for environkey in filter(lambda key: key in os.environ, environkeys):
+            del os.environ[environkey]
+        configfiles = [os.path.join(
+                       os.path.dirname(
+                           sys.modules[settings.__name__].__file__),
+                       'praw.ini'), 'praw.ini']
+        configdata = {}
+        for cfile in configfiles:
+            try:
+                with open(cfile, 'r') as f:
+                    configdata[f] = f.read()
+                    os.remove(f.name)
+            except IOError:
+                pass
+        try:
+            assertRaisesRegex(self, Exception, re.escape(str(configfiles)),
```

Comment:
I agree that it should be in practice something more specific. Though not being able to bump the major version number means that this cannot be changed.